src/utilities/compareVersions.js

```diff
@@ -0,0 +1,69 @@
+import { Client as BugsnagClient } from 'bugsnag-react-native';
+const bugsnagClient = new BugsnagClient();
+
+export function compareVersions(versionOne, versionTwo) {
+  try {
+    const result = versionToInteger(versionOne) - versionToInteger(versionTwo);
+    if (result > 0) return 1;
+    if (result < 0) return -1;
+    return 0;
+  } catch (error) {
+      // Errors thrown in migration method in dataMigration.js do not trickle down to bugsnag
+      // so will need manually notify bug snag.
+    error.message = `error verifying versions, versionOne: ${versionOne}, versionTwo ${versionTwo}`;
+    bugsnagClient.notify(error);
+  }
+  return 0;
+}
+// versionToInteger logic is the same as ./android/app/build.gradle
+function versionToInteger(version) {
+  const [majorMinorPatch, provisional] = version.split('-');
+  const [major, minor, patch] = majorMinorPatch.split('.');
+  return Number(major) * 10000000 +
```

Comment:
I think casting using `Number` is my favourite, but out of curiosity I went searching for why to use it over parseInt or parseFloat. Found a brilliant table on the matter!

https://stackoverflow.com/questions/17106681/parseint-vs-unary-plus-when-to-use-which/17106702#17106702

Some people debate the performance of each method, but from what I've read most discussion is naive as it's platform specific.

Reply:
Nice table, looking at it i just realised if someone there is no number after RC, Number('') will return zero, that should not be a problem or ?

Reply:
Shouldn't be `x + 0 = x` after all. More test cases! haha